### DISCUSSION TITLE
JSON schema for vgplot

### DISCUSSION THREAD

**Discussion by ilyabo** [uwdata/mosaic#179] [General]
Fantastic project! 

Do you have any plans for publishing a schema for vgplot specs? It would make it easier to create plots by enabling auto-completion and validation in editors. 

Maybe consider https://zod.dev/ for defining a schema which could be used for parsing and validation and would also serve as TypeScript types. JSON schema could then be generated from Zod definitions using [zod-to-json-schema](https://github.com/StefanTerdell/zod-to-json-schema)

**Comment by jheer** [comment]
Thanks for the suggestion! Yes, I would love to add TypeScript types and a JSON schema for the declarative specs. It's just a matter of finding sufficient time (or external contributions). Unfortunately I'm pretty tied up with other obligations until November.

**Reply by domoritz** [reply]
We can generate JSON schema from typescript using https://github.com/vega/ts-json-schema-generator.

**Reply by domoritz** [reply]
Done in https://github.com/uwdata/mosaic/pull/358.
